documentation/src/main/asciidoc/user_guide/clustering.adoc

```diff
@@ -269,58 +269,59 @@ While the primary is waiting for the confirmation message from the backup, reads
+node. This allows nodes to locate keys without making multicast requests or
+maintaining metadata for each key.
+
+The `numSegments` property configures the number of segments available.
+However, the number of segments cannot change unless the cluster is restarted.
+
+Likewise the key-to-segment mapping cannot change. Keys must always map to the
+same segments regardless of cluster topology changes. It is important that the
+key-to-segment mapping evenly distributes the number of segments allocated to each node while minimizing the number of segments that must move when the cluster topology changes.
+
+You can customize the key-to-segment mapping by configuring a
+link:https://docs.jboss.org/infinispan/{infinispanversion}/apidocs/org/infinispan/distribution/ch/KeyPartitioner.html[KeyPartitioner] or by using the
+link:#grouping_api[Grouping API].
+
+However, {brandname} provides the following implementations:
 
 SyncConsistentHashFactory::
-  An algorithm based on
-  link:http://en.wikipedia.org/wiki/Consistent_hashing[consistent hashing].
-  It always assigns a key to the same node in every cache as long as the cluster is
-  symmetric (i.e. all caches run on all nodes).
-  It does have some weaknesses: the load distribution is a bit uneven, and it also moves
-  more segments than strictly necessary on a join or leave.
-  Selected by default when server hinting is disabled.
+Uses an algorithm based on
+link:http://en.wikipedia.org/wiki/Consistent_hashing[consistent hashing]. Selected by default when server hinting is disabled.
++
+This implementation always assigns keys to the same nodes in every cache as
+long as the cluster is symmetric. In other words, all caches run on all nodes.
+This implementation does have some negative points in that the load distribution is slightly uneven. It also moves more segments than strictly necessary on a join or leave.
+
 TopologyAwareSyncConsistentHashFactory::
-  Similar to `SyncConsistentHashFactory`, but adapted for
-  link:#server_hinting[Server Hinting].
-  Selected by default when server hinting is enabled.
+Similar to `SyncConsistentHashFactory`, but adapted for
+link:#server_hinting[Server Hinting]. Selected by default when server hinting is enabled.
+
 DefaultConsistentHashFactory::
-  It achieves a more even distribution than `SyncConsistentHashFactory`, but it has one
-  disadvantage: the mapping of segments to nodes depends on the order in which caches
-  joined the cluster, so a key's owners are not guaranteed to be the same in all the
-  caches running in a cluster.
-  Used to be the default from version 5.2 to version 8.1 (with server hinting disabled).
+Achieves a more even distribution than `SyncConsistentHashFactory`, but with one
+disadvantage. The mapping of segments to nodes depends on the order in which
+caches join the cluster. As a result the owners of a key cannot be guaranteed
+to be the same in all the caches running in a cluster.
```

Comment:
This might need to be worded differently, it is quite confusing imo. I am guessing this is talking about the fact that a key may map to a different owner in a different cache, depending on the order of the cache topology.

Reply:
I believe there is an error in the original version: "the mapping of segments to nodes depends on the order in which caches joined the cluster". Shouldn't this be "the order in which nodes joined the cluster"?

Reply:
What you have sounds much better now :)